src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala

```diff
@@ -523,4 +529,120 @@ class WorkspaceApiServiceSpec extends FlatSpec with HttpService with ScalatestRo
+      }
+    Get(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/acl") ~>
+      addOpenAmCookie("owner-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK) {
+          status
+        }
+      }
+    Patch(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/acl", HttpEntity(ContentTypes.`application/json`, Seq.empty.toJson.toString)) ~>
+      addOpenAmCookie("owner-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK) { status }
+      }
+  }
+
+  it should "not allow an owner to lock a workspace with incomplete submissions" in withTestDataApiServices { services =>
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/lock") ~>
+      addOpenAmCookie("owner-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Conflict) { status }
+      }
+  }
+
+  it should "allow an owner to unlock the workspace" in withEmptyWorkspaceApiServices { services =>
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/lock") ~>
+      addOpenAmCookie("owner-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) { status }
+      }
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/unlock") ~>
+      addOpenAmCookie("owner-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) { status }
+      }
+  }
+
+  it should "not allow a non-owner to lock or unlock the workspace" in withEmptyWorkspaceApiServices { services =>
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/lock") ~>
+      addOpenAmCookie("write-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Forbidden) { status }
+      }
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/unlock") ~>
+      addOpenAmCookie("read-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.Forbidden) { status }
+      }
+  }
+
+  it should "not allow a no-access user to infer the existence the workspace by locking or unlocking" in withEmptyWorkspaceApiServices { services =>
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/lock") ~>
+      addOpenAmCookie("no-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NotFound) { status }
+      }
+    Put(s"/workspaces/${testData.workspace.namespace}/${testData.workspace.name}/unlock") ~>
+      addOpenAmCookie("no-access") ~>
+      sealRoute(services.workspaceRoutes) ~>
+      check {
+        assertResult(StatusCodes.NotFound) { status }
+      }
+  }
```

Comment:
Don't quite get it -- you feel we need a special test for this particular example of reading from a locked workspace?